packages/lesswrong/server/repos/VotesRepo.ts

```diff
@@ -109,4 +116,48 @@ export default class VotesRepo extends AbstractRepo<DbVote> {
       WHERE ARRAY_POSITION("authorIds", $1) IS NOT NULL
     `, [oldUserId, newUserId]);
   }
+
+  // Get votes from recent content by a user,
+  // to use to decide on their rate limit
+  // (note: needs to get the user's own self-upvotes so that
+  // it doesn't skip posts with no other votes)
+  async getVotesOnRecentContent(userId: string): Promise<RecentVoteInfo[]> {
+    const voteFields = `"Votes"._id, "Votes"."userId", "Votes"."power", "Votes"."documentId", "Votes"."collectionName", `
+    const votes = await this.any(`
+      (
+        SELECT ${voteFields} "Posts"."postedAt"
+        FROM "Votes"
+        JOIN "Posts" on "Posts"._id = "Votes"."documentId"
+        WHERE
+          "Votes"."documentId" in (
+            SELECT _id FROM "Posts" 
+            WHERE
+              "Posts"."userId" = '${userId}'
```

Comment:
This should be passed in as a parameter, not interpolated - SQL injection is a risk here